### DISCUSSION TITLE
legacy/new api

### DISCUSSION THREAD

**Discussion by michal-bartak** [Koenkk/zigbee2mqtt#8175] [General]
Hello
In zigbee2mqtt GUI, in Settings / Advanced, there is an item Legacy API (it's checked). 
I suppose it represents 
```
experimental:
  new_api: false
```
found in Home Assistant z2m integration configuration (however I didn't find it in `/share/zigbee2mqtt/configuration.yaml`)

I have two questions

1. what is this option for? Should I have disable legacy api?
2. not sure if is it correct place, but what is the relationship between settings found in integration configuration, `/share/zigbee2mqtt/configuration.yaml` and config shown in zigbee2mqtt GUI? Seams the same for most settings. 

with regards

**Comment by Koenkk** [comment]
- The legacy api is the `legacy_api` option (https://www.zigbee2mqtt.io/information/configuration.html)
- `experimental: new_api: false` can be removed (this option is not used anymore, new api is always enabled)
- with the GUI in the z2m frontend you can set settings in the `configuration.yaml` (its just an interface to configure this file)

**Reply by michal-bartak** [reply]
If you don't mind:
legacy_api: true disables enables legacy api? It's not completely clear from description.
```
# Optional: disables the legacy api (default: shown below)
  legacy_api: true
```

And what leaving legacy enabled is good for? Should I disable the legacy api for HomeAssistant?
Sorry for asking but found no detailed description.

**Reply by Koenkk** [reply]
`legacy_api: true` -> legacy api enabled, false = disabled. It doesn't matter for Home Assistant, this was left in to prevent a breaking change. For new installations it is disabled by default.

**Reply by michal-bartak** [reply]
I tried to disable legacy_api. It causes z2m GUI never finishing to load ending with some timeout message.

**Reply by Koenkk** [reply]
Zigbee2mqtt assistant or the builtin-frontend (https://www.zigbee2mqtt.io/information/frontend.html) ?

**Reply by michal-bartak** [reply]
> Zigbee2mqtt assistant or the builtin-frontend (https://www.zigbee2mqtt.io/information/frontend.html) ?

built-in frontend.
I'm using Home Assistant zigbee2mqtt official add-on.
While it's recent version of addon, HA version is from January: 2021.1.5
